### DISCUSSION TITLE
I need to run birt 4.10 in java 1.8.

### DISCUSSION THREAD

**Discussion by MuriloIPM** [eclipse-birt/birt#1272] [Q&A]
I need to run birt 4.10 in java 1.8, is this possible? I am aware that I would have to recompile the project

**Comment by ruspl-afed** [comment]
> I am aware that I would have to recompile the project

And before this, one will need to degrade settings and libraries back to Java 8. Could be quite an effort.

**Reply by MuriloIPM** [reply]
Thanks for the warning, my problem itself is as follows:
We have a report emitter with birt 4.4, using java 7 and running on a tomcat 7 server.
I updated to birt version 4.10 and java 17, but in many reports we use the following package in the `sun.misc.BASE64Decoder` script, I tried to recreate this class, but because of the module it didn't work, would you have any idea of a solution in java 17?
Observation: I cannot change the layouts to call the `java.util.base64.BASE64Decoder`, because the layout has to be compatible with old versions

**Reply by speckyspooky** [reply]
The only idea would be to create your own class which use internally the java.util.base64.BASE64Decoder and solve the dependency with the modules of Java 17. You can add the jar-file at your reports as a report-source-jar or you create your own plugin which you could add to the BIRT-plugins.
But I think in every case you cannot avoid to modify your reports. And under this conditions your question is what is better for your reports - stay in past with the older coding or change it to be perfomred for the future, pherhaps a step by step migration.

**Reply by MuriloIPM** [reply]
Sorry, but I don't know how to deal very well with modular java I managed to create this class but I got this error:
![image](https://user-images.githubusercontent.com/113636465/234373671-b1e18e3b-f986-4f59-90fa-7e16a2c5360b.png)

**Reply by wimjongman** [reply]
If you have these questions, then you will not be able to get it to work. Birt 4.10 will NOT work in Java 8. Please forget about it.
